editor/components/rich-text/inline-blocks.js

```diff
@@ -0,0 +1,135 @@
+/**
+ * WordPress dependencies
+ */
+import { Component, Fragment, compose } from '@wordpress/element';
+import { withSelect, withDispatch } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import InlineInsertionPoint from './inline-insertion-point';
+import MediaUpload from '../media-upload';
+
+class InlineBlocks extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.insert = this.insert.bind( this );
+		this.onSelectMedia = this.onSelectMedia.bind( this );
```

Comment:
As previously commented, I don't like the image specific logic here. Would be good to be part of the block registration somehow.

Reply:
Agreed. It's tricky because if the registered block returns a React component, RichText will render it to a string so it can't use React methods after being inserted. I've been contemplating whether it would be possible to use slot/fill or another method to render an inline block's "menu" (in this case the Media Library) either in the `Inserter` or `InlineBlocks` component, and then have the menu component return the element to be rendered to string on select, but this still doesn't seem ideal. 

Related: your https://github.com/WordPress/gutenberg/pull/5794#discussion_r187945949 
Also maybe related: the Edit toolbar https://github.com/WordPress/gutenberg/pull/5794#issuecomment-381516924 and how that could be rendered by React after insertion.